src/test/java/nextstep/subway/domain/SectionsTest.java

```diff
@@ -81,135 +154,160 @@ void setUp() {
+         * Then 추가가 된다
+         */
+        @DisplayName("기존 구간 맨 뒤에 새로운 구간을 추가 시 구간이 추가가 된다")
+        @Test
+        void 기존_구간_맨_뒤에_새로운_구간을_추가_시_구간이_추가가_된다() {
+            // Given
+            Station 판교역 = new Station("판교역");
 
-        // When && Then
-        assertThatThrownBy(() -> givenSections.addSection(신분당선, 강남역, 판교역, 10))
-            .isInstanceOf(IllegalDistanceSectionException.class);
-    }
+            // When
+            givenSections.addSection(신분당선, 양재역, 판교역, 4);
 
-    /**
-     * When 기존 구간에 상행역과 하행역 둘 중 하나도 포함되어있지 않으면
-     * Then 추가가 안된다
-     */
-    @DisplayName("기존 구간에 상행역과 하행역 둘 중 하나도 포함되어있지 않으면 추가가 안된다")
-    @Test
-    void 기존_구간에_상행역과_하행역_둘_중_하나도_포함되어있지_않으면_추가가_안된다() {
-        // Given
-        Station 판교역 = new Station("판교역");
-        Station 수지구청역 = new Station("수지구청역");
-
-        // When && Then
-        assertThatThrownBy(() -> givenSections.addSection(신분당선, 수지구청역, 판교역, 10))
-            .isInstanceOf(IllegalAddSectionException.class);
-    }
+            // Then
+            List<Section> sections = givenSections.getSections();
+            assertThat(sections.get(0).getUpStation()).isEqualTo(강남역);
+            assertThat(sections.get(0).getDownStation()).isEqualTo(양재역);
+            assertThat(sections.get(0).getDistance()).isEqualTo(10);
+            assertThat(sections.get(1).getUpStation()).isEqualTo(양재역);
+            assertThat(sections.get(1).getDownStation()).isEqualTo(판교역);
+            assertThat(sections.get(1).getDistance()).isEqualTo(4);
+        }
+
+        /**
+         * When 기존 구간과 동일한 구간 추가 요청시
+         * Then 추가가 안된다
+         */
+        @DisplayName("기존 구간과 동일한 구간 추가 요청시 추가가 안된다")
+        @Test
+        void 기존_구간과_동일한_구간_추가_요청시_추가가_안된다() {
+            // When && Then
+            assertThatThrownBy(() -> givenSections.addSection(신분당선, 강남역, 양재역, 4))
+                .isInstanceOf(DuplicateAddSectionException.class);
+        }
+
+        /**
+         * When 기존 구간에 기존 거리보다 새로운 구간의 거리가 같거나 더 클 시
+         * Then 추가가 안된다
+         */
+        @DisplayName("기존 구간에 기존 거리보다 새로운 구간의 거리가 같거나 더 클 시 추가가 안된다")
+        @Test
+        void 기존_구간에_기존_거리보다_새로운_구간의_거리가_더_클_시_추가가_안된다() {
+            // Given
+            Station 판교역 = new Station("판교역");
+
+            // When && Then
+            assertThatThrownBy(() -> givenSections.addSection(신분당선, 강남역, 판교역, 10))
+                .isInstanceOf(IllegalDistanceSectionException.class);
+        }
+
+        /**
+         * When 기존 구간에 상행역과 하행역 둘 중 하나도 포함되어있지 않으면
+         * Then 추가가 안된다
+         */
+        @DisplayName("기존 구간에 상행역과 하행역 둘 중 하나도 포함되어있지 않으면 추가가 안된다")
+        @Test
+        void 기존_구간에_상행역과_하행역_둘_중_하나도_포함되어있지_않으면_추가가_안된다() {
+            // Given
+            Station 판교역 = new Station("판교역");
+            Station 수지구청역 = new Station("수지구청역");
+
+            // When && Then
+            assertThatThrownBy(() -> givenSections.addSection(신분당선, 수지구청역, 판교역, 10))
+                .isInstanceOf(IllegalAddSectionException.class);
```

Comment:
다음과 같이 에러 메세지도 함께 검증해 줄 수도 있을 것 같아요 😄 
```suggestion
            assertThatThrownBy(() -> givenSections.addSection(신분당선, 수지구청역, 판교역, 10))
                    .isInstanceOf(IllegalAddSectionException.class)
                    .hasMessage("역을 추가 할 수 없습니다");
```